src/rpc/mining.cpp

```diff
@@ -818,6 +824,10 @@ static RPCHelpMan getblocktemplate()
     UniValue aRules(UniValue::VARR);
     aRules.push_back("csv");
     if (!fPreSegWit) aRules.push_back("!segwit");
+    if (consensusParams.signet_blocks) {
+        aRules.push_back("!signet");
```

Comment:
It would be useful to add a comment for non mining experts, such as yours truly, to explain what this does (especially the `!` part, which I guess doesn't mean "not" here). I found the discussion around `!segwit` rather confusing: https://github.com/bitcoin/bitcoin/pull/17946

Reply:
Added a comment